inc/integrations/class-jwt-auth.php

```diff
@@ -0,0 +1,157 @@
+<?php
+/**
+ * Previews.
+ *
+ * This functionality ensures that authenticated previews still work in Irving.
+ *
+ * @package WP_Irving
+ */
+
+namespace WP_Irving;
+
+class JWT_Auth {
+
+	/**
+	 * Cookie name that Irving core expects.
+	 *
+	 * @var string
+	 */
+	const COOKIE_NAME = 'authorizationBearerToken';
+
+	/**
+	 * Name for the keypair.
+	 *
+	 * @var string
+	 */
+	const KEYPAIR_NAME = 'wp-irving-jwt-auth';
+
+	/**
+	 * Cookie domain. Used to enable cross-domain cookie auth.
+	 *
+	 * @var string
+	 */
+	public $cookie_domain = '';
+
+	/**
+	 * Class instance.
+	 *
+	 * @var null|self
+	 */
+	protected static $instance;
+
+	/**
+	 * Get class instance
+	 *
+	 * @return self
+	 */
+	public static function instance() {
+		if ( ! isset( static::$instance ) ) {
+			static::$instance = new static();
+			static::$instance->setup();
+		}
+		return static::$instance;
+	}
+
+	/**
+	 * Setup the singleton.
+	 *
+	 * @todo Determine if we need to validate the token. This may not be needed
+	 * because the cookie is set to expire right before the auth token does.
+	 * Maybe we should hook into if/when the token is revoked so we can kill
```

Comment:
This is the hook you can use to do that: https://github.com/WP-API/jwt-auth/blob/ca76a9cd3e2a9bb5524188f710ab2ee959d7cd28/wp-includes/rest-api/auth/class-wp-rest-token.php#L406

This [hook](https://github.com/WP-API/jwt-auth/blob/develop/wp-includes/rest-api/auth/class-wp-rest-key-pair.php#L290-L307) is the one the plugin is using to revoke the token.

Reply:
Thanks - worked like a charm.